### PR TITLE
Revamp the package and drop unnecessary deps

### PR DESCRIPTION
**Pull request by sorawee**
The package would be significantly more useful if it doesn't have
unnecessary deps.

1. List test-version-case.rkt in compile-omit-paths, since we really do
   not want to have gui-lib, etc. as a dependency whether at run-time or
   build-time.

2. Drop pre-Racket support. We have already dropped the support of
   pre-Racket versions by naming files ending with rkt in 9fdc2703bf4.

3. Switch mzscheme and #lang scheme to #lang racket

4. Switch SRFI usage to pure Racket

5. Rely on syntax/parse to report a parse error.

6. Correct documentation

7. Add some tests

### CODE REVIEW

File: version-case.scrbl [hunk 1]
```diff
@@ -25,11 +25,11 @@ with both the old and new @racket[unit] libraries.
 
 @racketblock[
 (module some-sample-unit-code mzscheme
-  (require (planet "version-case.ss" ("dyoo" "version-case.plt" 1))
+  (require version-case
            (lib "mred.ss" "mred"))
   
-  ;; Small test code to see that we can write unit-dependent code
-  ;; that still runs under both 360 and 369.
```
Comment (samth) [inline version-case.scrbl:32]: Dropping pre-racket support means that we can't do what this comment says.

Comment (sorawee) [reply version-case.scrbl:32]: Do you have any other good example?

Comment (sorawee) [reply version-case.scrbl:32]: I changed the example.

**Comment by sorawee** [comment]
Can this be merged? I'd love to use it soon, but only when the deps are already trimmed.

**Comment by samth** [comment]
Can you add something in the docs/readme about what versions are intended to be supported?

**Comment by sorawee** [comment]
I added a remark that it's supported for 6.0 and later.

**Pull request merged**